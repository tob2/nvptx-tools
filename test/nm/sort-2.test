Test sorting options, object files linked together (in contrast to 'sort-1.test').


Link.

RUN: %target_ld_cmd -o %t.nvptx --ignore-unresolved-symbol __mprec_bigtens %S/libc/libc_a-dtoa.o %S/libc/libc_a-environ.o %S/libc/libc_a-memset.o


Default.

RUN: %env_LC_COLLATE=C_target_nm_cmd %t.nvptx > %t.stdout 2> %t.stderr
RUN: cmp %S/sort-2.nm.golden %t.stdout
RUN: ! test -s %t.stderr


'--reverse-sort'/'-r'.

RUN: %env_LC_COLLATE=C_target_nm_cmd --reverse-sort %t.nvptx > %t.stdout 2> %t.stderr
RUN: cmp %S/sort-2.nm_-r.golden %t.stdout
RUN: ! test -s %t.stderr

RUN: %env_LC_COLLATE=C_target_nm_cmd -r %t.nvptx > %t.stdout 2> %t.stderr
RUN: cmp %S/sort-2.nm_-r.golden %t.stdout
RUN: ! test -s %t.stderr

RUN: %env_LC_COLLATE=C_target_nm_cmd -r --reverse-sort %t.nvptx > %t.stdout 2> %t.stderr
RUN: cmp %S/sort-2.nm_-r.golden %t.stdout
RUN: ! test -s %t.stderr


'--numeric-sort'/'-n'.

RUN: %env_LC_COLLATE=C_target_nm_cmd --numeric-sort %t.nvptx > %t.stdout 2> %t.stderr
RUN: cmp %S/sort-2.nm_-n.golden %t.stdout
RUN: ! test -s %t.stderr

RUN: %env_LC_COLLATE=C_target_nm_cmd -n %t.nvptx > %t.stdout 2> %t.stderr
RUN: cmp %S/sort-2.nm_-n.golden %t.stdout
RUN: ! test -s %t.stderr

RUN: %env_LC_COLLATE=C_target_nm_cmd -n --numeric-sort %t.nvptx > %t.stdout 2> %t.stderr
RUN: cmp %S/sort-2.nm_-n.golden %t.stdout
RUN: ! test -s %t.stderr


'--numeric-sort'/'-n', '--reverse-sort'/'-r'.

RUN: %env_LC_COLLATE=C_target_nm_cmd -n --reverse-sort %t.nvptx > %t.stdout 2> %t.stderr
RUN: cmp %S/sort-2.nm_-nr.golden %t.stdout
RUN: ! test -s %t.stderr

RUN: %env_LC_COLLATE=C_target_nm_cmd -nr %t.nvptx > %t.stdout 2> %t.stderr
RUN: cmp %S/sort-2.nm_-nr.golden %t.stdout
RUN: ! test -s %t.stderr

RUN: %env_LC_COLLATE=C_target_nm_cmd -nrnn %t.nvptx > %t.stdout 2> %t.stderr
RUN: cmp %S/sort-2.nm_-nr.golden %t.stdout
RUN: ! test -s %t.stderr

RUN: %env_LC_COLLATE=C_target_nm_cmd --numeric-sort --reverse-sort %t.nvptx > %t.stdout 2> %t.stderr
RUN: cmp %S/sort-2.nm_-nr.golden %t.stdout
RUN: ! test -s %t.stderr


'--no-sort'/'-p'.

RUN: sort < %S/sort-2.nm.golden > %t.sort-2.nm.golden.sorted

RUN: %env_LC_COLLATE=C_target_nm_cmd --no-sort %t.nvptx > %t.stdout 2> %t.stderr
RUN: sort < %t.stdout > %t.stdout.sorted
RUN: cmp %t.sort-2.nm.golden.sorted %t.stdout.sorted
RUN: ! test -s %t.stderr

RUN: %env_LC_COLLATE=C_target_nm_cmd -p %t.nvptx > %t.stdout 2> %t.stderr
RUN: sort < %t.stdout > %t.stdout.sorted
RUN: cmp %t.sort-2.nm.golden.sorted %t.stdout.sorted
RUN: ! test -s %t.stderr

RUN: %env_LC_COLLATE=C_target_nm_cmd --no-sort -p %t.nvptx > %t.stdout 2> %t.stderr
RUN: sort < %t.stdout > %t.stdout.sorted
RUN: cmp %t.sort-2.nm.golden.sorted %t.stdout.sorted
RUN: ! test -s %t.stderr
