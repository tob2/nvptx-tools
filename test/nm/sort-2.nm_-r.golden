0000000000000000 t quorem
0000000000000000 T memset
0000000000000000 d initial_env
0000000000000000 D environ
0000000000000000 T _dtoa_r
                 U __mprec_bigtens
0000000000000000 d $LC2
